lib/css/CssModulesPlugin.js

```diff
@@ -358,8 +361,36 @@ class CssModulesPlugin {
 							return new CssModule(createData);
 						});
 				}
+
+				JavascriptModulesPlugin.getCompilationHooks(
+					compilation
+				).renderModuleContent.tap(PLUGIN_NAME, (source, module) => {
+					if (module instanceof CssModule && module.hot) {
+						const exports = module.buildInfo.cssData.exports;
+						const stringifiedExports = JSON.stringify(
+							Array.from(exports).reduce((obj, [key, value]) => {
+								obj[key] = value;
+								return obj;
+							}, {})
+						);
+
+						const hmrCode = Template.asString([
+							"",
+							`var exports = ${stringifiedExports};`,
```

Comment:
```suggestion
							`var exports = ${JSON.stringify(stringifiedExports)};`,
```

Should compare string or object value strings

This fixes infinite recursive execution issue above

Reply:
Fixed, good catch